sphinx/builders/linkcheck.py

```diff
@@ -630,21 +633,6 @@ class RateLimit(NamedTuple):
     next_check: float
 
 
-def rewrite_github_anchor(app: Sphinx, uri: str) -> str | None:
```

Comment:
Please note: some people may still be using `rewrite_github_anchor` in their Sphinx project configurations.  We should add a deprecation notice to the function instead of removing it completely, so that they have time to adjust.  Features are removed two major versions after they are first indicated for removal (so Sphinx v10.0 for this).

More information about this can be found in Sphinx's deprecation policy documentation: https://www.sphinx-doc.org/en/master/internals/release-process.html#deprecation-policy

Reply:
I've updated the PR to add a deprecation message to `rewrite_github_anchor` instead of deleting it.